src/System.Net.Security/src/System/Net/Security/SslConnectionInfo.OSX.cs

```diff
@@ -60,85 +91,797 @@ private struct TlsMapping
+                TlsMapping.EcDhStatic(CipherAlgorithmType.Aes128, HashAlgorithmType.Sha256)
+            },
+
+            {
+                TlsCipherSuite.TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384,
+                TlsMapping.EcDhStatic(CipherAlgorithmType.Aes256, HashAlgorithmType.Sha384)
+            },
+
+            {
+                TlsCipherSuite.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,
+                TlsMapping.EcDhEphem(CipherAlgorithmType.Aes128, HashAlgorithmType.Sha256)
             },
 
             {
                 TlsCipherSuite.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384,
-                TlsMapping.EcDhe(CipherAlgorithmType.Aes256, HashAlgorithmType.Sha384)
+                TlsMapping.EcDhEphem(CipherAlgorithmType.Aes256, HashAlgorithmType.Sha384)
+            },
+
+            {
+                TlsCipherSuite.TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256,
+                TlsMapping.EcDhStatic(CipherAlgorithmType.Aes128, HashAlgorithmType.Sha256)
+            },
+
+            {
+                TlsCipherSuite.TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384,
+                TlsMapping.EcDhStatic(CipherAlgorithmType.Aes256, HashAlgorithmType.Sha384)
+            },
+
+            {
+                TlsCipherSuite.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+                TlsMapping.EcDhEphem(CipherAlgorithmType.Aes128, HashAlgorithmType.Sha256)
             },
 
             {
                 TlsCipherSuite.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
-                TlsMapping.EcDhe(CipherAlgorithmType.Aes256, HashAlgorithmType.Sha384)
+                TlsMapping.EcDhEphem(CipherAlgorithmType.Aes256, HashAlgorithmType.Sha384)
             },
 
             {
+                TlsCipherSuite.TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256,
+                TlsMapping.EcDhStatic(CipherAlgorithmType.Aes128, HashAlgorithmType.Sha256)
+            },
+
+            {
+                TlsCipherSuite.TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384,
+                TlsMapping.EcDhStatic(CipherAlgorithmType.Aes256, HashAlgorithmType.Sha384)
+            },
+            {
                 TlsCipherSuite.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
-                TlsMapping.EcDhe(CipherAlgorithmType.Aes128, HashAlgorithmType.Sha256)
+                TlsMapping.EcDhEphem(CipherAlgorithmType.Aes128, HashAlgorithmType.Sha256)
             },
 
             {
                 TlsCipherSuite.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
-                TlsMapping.EcDhe(CipherAlgorithmType.Aes256, HashAlgorithmType.Sha384)
+                TlsMapping.EcDhEphem(CipherAlgorithmType.Aes256, HashAlgorithmType.Sha384)
+            },
+
+            {
+                TlsCipherSuite.TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256,
+                TlsMapping.EcDhStatic(CipherAlgorithmType.Aes128, HashAlgorithmType.Sha256)
+            },
+
+            {
+                TlsCipherSuite.TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384,
+                TlsMapping.EcDhStatic(CipherAlgorithmType.Aes256, HashAlgorithmType.Sha384)
```

Comment:
Do we have tests that use all of these?  So much parameterization, seems like it'd be easy to fat-finger one of these entries and cause a subtle issue.